Stack/Core/Stack/State/NodeState.cs

```diff
@@ -2345,11 +2333,13 @@ public virtual void ReportEvent(ISystemContext context, IFilterTarget e)
                 {
                     RemoveReference(referenceTypeId, isInverse, target.NodeId);
                 }
-
+                    
                 if (entry == null)
                 {
                     entry = new Notifier();
-                    m_notifiers.Add(entry);
+                    Notifier[] notifiers = new Notifier[m_notifiers.Length + 1];
```

Comment:
Ooops! You have to also copy the contents from the existing m_notifiers to the new one.